pkg/pillar/evetpm/tpm.go

```diff
@@ -802,6 +853,112 @@ func pcrBankSHA256EnabledHelper() bool {
 	return err == nil
 }
 
+func getTpmEventLogFiles(evtDir string, evtFile string) ([]string, error) {
+	files, err := filepath.Glob(evtDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to enumerate tpm(s)")
+	}
+
+	tpmEvtFiles := make([]string, 0)
+	for _, path := range files {
+		evtBin := filepath.Join(path, evtFile)
+		if fileutils.FileExists(nil, evtBin) {
+			tpmEvtFiles = append(tpmEvtFiles, evtBin)
+		}
+	}
+
+	return tpmEvtFiles, nil
+}
+
+func collectPrevTpmEventLogs(evtLogSealing string, evtLogFailing string) error {
+	// just call it to count available tpm devices
+	paths, err := getTpmEventLogFiles(TPMEvtLogDir, TpmEvtLogBinFile)
+	if err != nil || len(paths) == 0 {
+		return fmt.Errorf("failed to find any TPM event log file: %w", err)
+	} else if len(paths) == 0 {
+		return fmt.Errorf("failed to find any TPM event log file")
+	}
+
+	saveFailed := len(paths)
```

Comment:
leftToSave would make more sense with this logic. Or a saveFailed initialized to zero and incremented on failure.